Realm/Realm.Unity/Editor/Common/RealmWeaver.cs

```diff
@@ -1504,6 +1504,31 @@ private TypeDefinition WeaveRealmObjectHelper(TypeDefinition realmObjectType, Me
             return helperType;
         }
 
+        private bool IsUsingSync()
+        {
+            try
+            {
+                return IsMethodUsed(_references.SyncConfiguration);
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private bool IsMethodUsed(TypeReference type)
+        {
+            _ = type ?? throw new ArgumentNullException(nameof(type));
```

Comment:
[nit] We don't validate arguments on internal methods and instead rely on our epic coding skills to avoid introducing bugs 😄